include/aikido/control/BarrettHandPositionCommandExecutor.hpp

```diff
@@ -1,86 +1,28 @@
 #ifndef AIKIDO_CONTROL_BARRETTHANDPOSITIONCOMANDEXECUTOR_HPP_
 #define AIKIDO_CONTROL_BARRETTHANDPOSITIONCOMANDEXECUTOR_HPP_
-#include "BarrettFingerPositionCommandExecutor.hpp"
-#include "BarrettFingerSpreadCommandExecutor.hpp"
-#include <dart/collision/CollisionDetector.hpp>
-#include <dart/collision/CollisionOption.hpp>
-#include <dart/collision/CollisionGroup.hpp>
-#include <dart/collision/CollisionFilter.hpp>
-#include <dart/dynamics/dynamics.hpp>
 #include <Eigen/Dense>
 #include <future>
-#include <mutex>
-#include <chrono>
 
 namespace aikido {
 namespace control {
 
-/// Position command executor for BarrettHand fingers. 
-/// Assumes that fingers are underactuated: proximal joint is actuated
-/// and distal joint moves with certain mimic ratio until collision.
-/// If 
+/// Abstract class for position commands for a Barrett Hand.
 class BarrettHandPositionCommandExecutor
 {
 public:
-  /// Constructor.
-  /// \param _positionCommandExecutors 3 executors to control
-  ///        proximal and distal joints of the fingers. The first two fingers
-  ///        should have (spread, proximal, distal) joints. 
-  ///        Third finger should have 2 joints (proximal, distal).
-  /// \param _spreadCommandExecutor Executors to control
-  ///        spreads of the fingers.
-   BarrettHandPositionCommandExecutor(
-    std::array<BarrettFingerPositionCommandExecutorPtr, 3> _positionCommandExecutors,
-    BarrettFingerSpreadCommandExecutorPtr _spreadCommandExecutor);
+  virtual ~BarrettHandPositionCommandExecutor() = default;
 
-  /// Set relevant variables for moving fingers.
-  /// In order to move the fingers, step method should be called multiple times 
-  /// until future returns.
-  /// \param _goalPositions End dof pose for proximal joints and spread. 
-  ///        First 3 should be for proximal joints, the last element should be
-  ///        for spread. If _positions are above/below joint limits,
-  ///        the fingers will move only upto the limit.
-  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// Execute hand to goalPosition
+  /// \param goalPositions End dof pose for proximal joints and spread.
   /// \return Future which becomes available when the execution completes.
-  std::future<void> execute(
-    Eigen::Matrix<double, 4, 1> _goalPositions, 
-    ::dart::collision::CollisionGroupPtr _collideWith);
+  virtual std::future<void> execute(
+    Eigen::Matrix<double, 4, 1> goalPositions) = 0;
```

Comment:
Actually, make this an `Eigen::VectorXd` since this is now a generic interface - potentially for hardware other than the BarrettHand. E.g. we _might_ decide to use a similar interface for the pan-tilt unit.

Also, [passing a fixed-size `Eigen::Matrix` by value is **wrong**](https://eigen.tuxfamily.org/dox/group__TopicPassingByValue.html) because the argument may have incorrect alignment. This is more than a nit. 😉